app/services/hearing_datetime_service.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# HearingDatetimeService is intended to replace HearingTimeService by taking advantage of the new hearings table columns
+# scheduled_in_timezone and (for AMA hearings) scheduled_datetime. The two services will work in parallel until
+# hearings with nil values for those columns have been held, at which point we should be able to deprecate
+# HearingTimeService as well as the hearings.scheduled_time column (which is unable to store the hearing date due
+# to it's datatype in postgres). Until then, this class will work in parallel to HearingTimeService through the
+# facade hearing.time as set in the module HearingTimeConcern.rb
+class HearingDatetimeService
+  class << self
+    # Combines a date and time string to create a Time object.
+    #
+    # @param date [Date] a Date object for which a hearing will be scheduled.
+    # @param time_string [String] a formatted string with scheduling details, `12:00 PM Eastern Time (US & Canada)`.
+    # @return [Time] the Time object in the calculated time zone and DST offset.
+    def prepare_time_for_storage(date:, time_string:)
+      return nil unless date && time_string
+
+      time_without_zone = time_string.split(" ", 3).take(2).join(" ")
+      "#{date} #{time_without_zone}".in_time_zone(timezone_from_time_string(time_string))
+    end
+
+    private
+
+    # Converts time zone from the scheduled_time_string into a region string recognized by
+    # Rails ActiveSupport::Timezone
+    #
+    # @param scheduled_time_string [String] Formatted time and timezone string, `12:00 PM Eastern Time (US & Canada)`.
+    # @return [String] e.g. `America/New_York`
+    def timezone_from_time_string(scheduled_time_string)
+      time_str_split = scheduled_time_string.split(" ", 3)
+
+      tz_str = ActiveSupport::TimeZone::MAPPING[time_str_split[2]]
+      tz_str = ActiveSupport::TimeZone::MAPPING.key(time_str_split[2]) if tz_str.nil?
```

Comment:
```suggestion
      tz_str = ActiveSupport::TimeZone[time_str_split[2]]
```
The `[]` function is defined so you don't need to directly access the MAPPING object.

Reply:
@ThorntonMatthew I can't remember why we added the reverse condition on line 35. Can you think of an edge case? I'm thinking we may have just sniped this logic from some other method where that was a possibility. 🤔

Reply:
I think it was in case we passed in "America/New_York" for some reason (a value in the MAPPING hash versus the key).